Mathlib/Data/Nat/Find.lean

```diff
@@ -97,6 +97,11 @@ variable [DecidablePred q] in
 lemma find_mono (h : ∀ n, q n → p n) {hp : ∃ n, p n} {hq : ∃ n, q n} : Nat.find hp ≤ Nat.find hq :=
   Nat.find_min' _ (h _ (Nat.find_spec hq))
 
+variable [DecidablePred q] in
+lemma find_congr {hp : ∃ n, p n} {hq : ∃ n, q n} (hpq : ∀ {n}, p n ↔ q n) :
```

Comment:
Is there a difference with simply writing:
```suggestion
lemma find_congr [DecidablePred q] {hp : ∃ n, p n} {hq : ∃ n, q n} (hpq : ∀ {n}, p n ↔ q n) :
```

Reply:
Also, should this be tagged `@[congr]`?

Reply:
I had it written with `variable in` since that's what the lemma above looked like:
```
variable [DecidablePred q] in
lemma find_mono (h : ∀ n, q n → p n) {hp : ∃ n, p n} {hq : ∃ n, q n} : Nat.find hp ≤ Nat.find hq :=
  Nat.find_min' _ (h _ (Nat.find_spec hq))
```

Reply:
With or without `@[congr`], usage looks the same when using the `congr` tactic. I don't think I understand what `@[congr]` is for.
```
example [DecidablePred q] {hp : ∃ n, p n} {hq : ∃ n, q n} (hpq : ∀ {n}, p n ↔ q n) :
    Nat.find hp = Nat.find hq := by
  congr
  guard_target = p = q
  ext
  exact hpq
```

Reply:
> I don't think I understand what @[congr] is for.

To be honest, after trying out a few things I'm not too sure either. I suppose we'll do without `@[congr]` then!